packages/scene-composer/src/components/StateManager.tsx

```diff
@@ -216,6 +232,38 @@ const StateManager: React.FC<SceneComposerInternalProps> = ({
     }
   }, [sceneContentUri]);
 
+  useEffect(() => {
+    if (sceneMetadataModule && matterportModelId) {
+      sceneMetadataModule
+        .getSceneInfo()
+        .then((sceneInfo) => {
+          if (sceneInfo && sceneInfo.generatedSceneMetadata) {
+            const accessToken = sceneInfo.generatedSceneMetadata[MATTERPORT_ACCESS_TOKEN];
+            const applicationKey = sceneInfo.generatedSceneMetadata[MATTERPORT_APPLICATION_KEY];
+
+            let updatedExternalLibraryConfig = { ...externalLibraryConfig };
```

Comment:
this is a shallow clone so when you do `updatedExternalLibraryConfig.matterport.modelId`, it's still updating the original one. you should do a deep clone, or just making the variable for matterport config only. something like this
```
const updatedMatterportLibraryConfig = { ...externalLibraryConfig?.matterport };
updatedMatterportLibraryConfig.modelId = matterportModelId;
updatedMatterportLibraryConfig.accessToken = accessToken;
updatedMatterportLibraryConfig.applicationKey = applicationKey;
setUpdatedExternalLibraryConfig({ ...externalLibraryConfig, matterport: updatedMatterportLibraryConfig });
```